src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -481,21 +481,14 @@ public FleetStateStats update() {
             }
         }
 
-        // fleet could be updated outside of plugin, we should be ready that
-        // real target capacity is zero or less then plugin thinks and make sure
-        // new target capacity will not be negative
-        final int targetCapacity = Math.max(0,
-                currentState.getNumDesired() - currentInstanceIdsToTerminate.size() + currentToAdd);
-        currentState = new FleetStateStats(currentState, targetCapacity);
-
-        updateByState(currentToAdd, currentInstanceIdsToTerminate, targetCapacity, currentState);
+        FleetStateStats updatedState = updateByState(currentToAdd, currentInstanceIdsToTerminate, currentState);
```

Comment:
I guess we are updating number of active instances too so probably creating new object instead of just updating targetCapacity. We should probably just re-use currentState object to point to the newer state object and let updateByState return FleetStateStats.